app/test/unit/git/rebase/progress-test.ts

```diff
@@ -100,6 +127,38 @@ describe('git/rebase', () => {
       expect(result).toBe(RebaseResult.ConflictsEncountered)
     })
 
+    it('reported step-by-step progress before encountering conflicts', () => {
+      expect(progress).toEqual([
+        {
+          currentCommitSummary: 'Feature Branch First Commit!',
+          kind: 'rebase',
+          rebasedCommitCount: 1,
+          title: 'Rebasing commit 1 of 10 commits',
+          totalCommitCount: 10,
+          value: 0.1,
+        },
+      ])
+    })
+
+    it('reports progress after resolving conflicts', async () => {
+      const strategy = ManualConflictResolutionKind.theirs
+      const progressCb = (p: IRebaseProgress) => progress.push(p)
+
+      while (result === RebaseResult.ConflictsEncountered) {
+        result = await resolveAndContinue(repository!, strategy, progressCb)
+      }
+
+      expect(progress.length).toEqual(10)
+      expect(progress[9]).toEqual({
+        currentCommitSummary: 'Feature Branch Tenth Commit!',
+        kind: 'rebase',
+        rebasedCommitCount: 10,
+        title: 'Rebasing commit 10 of 10 commits',
+        totalCommitCount: 10,
+        value: 1,
+      })
```

Comment:
Something I find quite useful on situations where there is a large object to assert is to use [jest snapshots](https://jestjs.io/docs/en/snapshot-testing).

There's some controversy around jest snapshots and I think it's relatively easy to abuse them (e.g by having huge snapshots with tons of data that can't easily be reviewed and changes often), but when I saw this assertion I thought about them :)

To be clear, I'm not suggesting changing the PR to use them here (there are no prior art of snapshots in Desktop so probably there needs to be a more in-depth discussion/investigation before introducing them), just want to showcase them just in case you don't know about them already 😃

Reply:
Yeah, I did consider snapshots here and I have nothing against them personally but I opted not to for the exact reason you mentioned, i.e. lack of prior art.